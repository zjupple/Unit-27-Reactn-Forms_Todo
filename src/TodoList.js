import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

function TodoList() {
  const [todos, setTodos] = useState([]);

  // add a new todo
  const create = newTodo => {
    setTodos(todos => [...todos, newTodo]);
  };

  // update a todo with updatedTask
  // for loop with alert of non edited task here
  const update = (id, updatedTask) => {
    setTodos(todos =>
      todos.map(todo =>
        todo.id === id ? { ...todo, task: updatedTask } : todo
      )
    );
  };

  // delete a todo by id
  // can use a for loop with condition here
  const remove = id => {
    setTodos(todos => todos.filter(todo => todo.id !== id));
  };


  // can use props here

  const todoComponents = todos.map(todo => (
    <Todo
      remove={remove}
      key={todo.id}
      id={todo.id}
      task={todo.task}
      update={update}
    />
  ));

  // prop created too pass down into newtodo form

  return (
    <div>
      <NewTodoForm createTodo={create} />
      <ul>{todoComponents}</ul>
    </div>
  );
}

export default TodoList;
