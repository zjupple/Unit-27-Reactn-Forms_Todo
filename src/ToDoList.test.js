import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import TodoList from "./TodoList";



it("renders without crashing", function() {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});