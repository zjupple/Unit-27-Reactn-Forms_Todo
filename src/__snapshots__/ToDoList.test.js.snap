// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="todo"
        >
          Task:
        </label>
        <input
          id="todo"
          name="todo"
          type="text"
          value=""
        />
        <button
          class="toDoButton"
        >
          Add a todo!
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
